src/node/createSetupServer.ts

```diff
@@ -168,6 +173,10 @@ export function createSetupServer(...interceptors: Interceptor[]) {
       },
 
       resetHandlers(...nextHandlers) {
+        // Clear pending requests so they don't resolve
+        Object.values(pendingRequests).forEach(clearTimeout)
```

Comment:
Technically, we are only clearing the timeouts that resolve the responses. However, that still leaves the requests pending, as what it means is that the request middleware just never resolves, as opposed to _aborts_ a request. 

I think if we wish to support this in NodeJS we need to extend `node-request-interceptor` to provide a client-agnostic API to abort requests. Both `ClientRequest` and `XMLHttpRequest` should support aborting.

Reply:
`setupWorker` has the same issue, but since in that case, the loss of the context means closing a tab, unresolved Promise will be dumped and forgotten (at least, I hope it would be). In NodeJS that's not the case, as the interception happens on the entire process.

Reply:
Yea, that's also probably why I needed the extra check in `setupServer` and not in `setupWorker`.

```ts
if (pendingRequests.has(mockedRequest.id)) {
                resolve(mockedResponse)
                pendingRequests.delete(mockedRequest.id)
}
```